app_libpng_test/example.c

```diff
@@ -0,0 +1,1052 @@
+         * not necessary when reading the image, because the alpha channel is
+         * preserved; if it were to be removed, for example if we requested
+         * PNG_FORMAT_RGB, then either a solid background color would have to
+         * be supplied, or the output buffer would have to be initialized to
+         * the actual background of the image.
+         *
+         * The fourth argument to png_image_finish_read is the 'row_stride' -
+         * this is the number of components allocated for the image in each
+         * row.  It has to be at least as big as the value returned by
+         * PNG_IMAGE_ROW_STRIDE, but if you just allocate space for the
+         * default, minimum size, using PNG_IMAGE_SIZE as above, you can pass
+         * zero.
+         *
+         * The final argument is a pointer to a buffer for the colormap;
+         * colormaps have exactly the same format as a row of image pixels
+         * (so you choose what format to make the colormap by setting
+         * image.format).  A colormap is only returned if
+         * PNG_FORMAT_FLAG_COLORMAP is also set in image.format, so in this
+         * case NULL is passed as the final argument.  If you do want to force
+         * all images into an index/color-mapped format, then you can use:
+         *
+         *    PNG_IMAGE_COLORMAP_SIZE(image)
+         *
+         * to find the maximum size of the colormap in bytes.
+         */
+      
+      if (buffer != NULL &&
+         png_image_finish_read(&image, NULL/*background*/, buffer,
+               0/*row_stride*/, NULL/*colormap*/) != 0)
+      {
+         /* Now write the image out to the second argument.  In the write
+            * call 'convert_to_8bit' allows 16-bit data to be squashed down to
+            * 8 bits; this isn't necessary here because the original read was
+            * to the 8-bit format.
+            */
+         
+         if (png_image_write_to_file(&image, second_arg, 0/*convert_to_8bit*/,
+               buffer, 0/*row_stride*/, NULL/*colormap*/) != 0)
+         {
+            /* The image has been written successfully. */
+            return 0;
+         }
+      }
+      else
+      {
+         /* Calling png_image_free is optional unless the simplified API was
+            * not run to completion.  In this case, if there wasn't enough
+            * memory for 'buffer', we didn't complete the read, so we must
+            * free the image:
+            */
+         if (buffer == NULL)
+            png_image_free(&image);
+         else
+            free(buffer);
+      }
+      
+
+      /* Something went wrong reading or writing the image.  libpng stores a
+       * textual message in the 'png_image' structure:
+       */
+      fprintf(stderr, "pngtopng: error: %s\n", image.message);
+      exit(1);
+   }
+
+   fprintf(stderr, "pngtopng: usage: pngtopng input-file output-file\n");
+   exit(2);
+
+   return 0;
+}
+
+#if 0 /* in case someone actually tries to compile this */
```

Comment:
This code is redundant (L150-L1052), please remove it from the test suite when you organise it into `uktest`, thanks!